### ISSUE TITLE
Notifications leak Byobu's private discussions if the tag is followed

### ISSUE THREAD

**Issue by imorland** [FriendsOfFlarum/follow-tags#5]
When both fof/follow-tags and fof/byobu are enabled, a private discussion under a tag which is followed by users is notified, and therefore leaks the content (and existence) of that discussion.

**Comment by datitisev** [comment]
Hm, looks like the logic I used isn't hooked into by byobu. I should check for private as well then, and act accordingly.

**Comment by imorland** [comment]
This is on 0.3.0 - had the same behaviour on 0.2.0 as well (never upgraded to 0.2.1)

For the moment, we are using this patch as a workaround

```
diff --git a/src/Jobs/SendNotificationWhenDiscussionIsStarted.php b/src/Jobs/SendNotificationWhenDiscussionIsStarted.php
index dbc3fa2..6d85be4 100644
--- a/src/Jobs/SendNotificationWhenDiscussionIsStarted.php
+++ b/src/Jobs/SendNotificationWhenDiscussionIsStarted.php
@@ -44,7 +44,7 @@ class SendNotificationWhenDiscussionIsStarted implements ShouldQueue
         $tags = $this->discussion->tags;
         $tagIds = $tags->map->id;
 
-        if ($tags->isEmpty()) {
+        if ($tags->isEmpty() || $this->discussion->is_private) {
             return;
         }
 
diff --git a/src/Jobs/SendNotificationWhenReplyIsPosted.php b/src/Jobs/SendNotificationWhenReplyIsPosted.php
index 3d0e040..2295918 100644
--- a/src/Jobs/SendNotificationWhenReplyIsPosted.php
+++ b/src/Jobs/SendNotificationWhenReplyIsPosted.php
@@ -51,7 +51,7 @@ class SendNotificationWhenReplyIsPosted implements ShouldQueue
         $tags = $discussion->tags;
         $tagIds = $tags->map->id;
 
-        if ($tags->isEmpty()) {
+        if ($tags->isEmpty() || $discussion->is_private) {
             return;
         }
 

```

**Comment by datitisev** [comment]
Fixed with 0.3.1 - uses `whereVisibleTo` now.
